core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -3185,16 +3217,103 @@ class DAGSchedulerSuite extends SparkFunSuite with TempLocalSparkContext with Ti
+    this.dagSchedulerInterceptor = new DagSchedulerInterceptor {
+      override def interceptHandleTaskCompletion(event: CompletionEvent): Unit = {
+        event.reason match {
+          case Success if event.task.isInstanceOf[ResultTask[_, _]] =>
+            assert(resubmitFailedStageReached(0))
+            monitor.synchronized {
+              monitor.notify()
+            }
+
+          case _ =>
+        }
+      }
+
+      override def interceptResubmitFailedStages(): Unit = {
+        monitor.synchronized {
+          resubmitFailedStageReached(0) = true
+          monitor.notify()
+          monitor.wait()
+        }
+      }
+    }
+
+    this.beforeEach()
+
+    val numPartitions = 2
+    val (shuffleId1, shuffleId2) = constructTwoIndeterminateStage()
+    completeShuffleMapStageSuccessfully(shuffleId2, 0, numPartitions)
+    val resultStage = scheduler.stageIdToStage(2).asInstanceOf[ResultStage]
+    val activeJob = resultStage.activeJob
+    assert(activeJob.isDefined)
+    // The result stage is still waiting for its 2 tasks to complete
+    assert(resultStage.findMissingPartitions() == Seq.tabulate(numPartitions)(i => i))
+    new Thread(() => {
+      runEventInCurrentThread(
+        makeCompletionEvent(
+          taskSets(2).tasks(0),
+          FetchFailed(makeBlockManagerId("hostA"), shuffleId1, 0L, 0, 0, "ignored"),
+          null))
+    }).start()
+
+    monitor.synchronized {
+      if (!resubmitFailedStageReached(0)) {
+        monitor.wait()
+      }
+    }
+    assert(resubmitFailedStageReached(0))
+    new Thread(() => {
+      runEventInCurrentThread(makeCompletionEvent(taskSets(2).tasks(1), Success, 11))
+    }).start()
+
+    val shuffleStage1 = this.scheduler.shuffleIdToMapStage(shuffleId1)
+    val shuffleStage2 = this.scheduler.shuffleIdToMapStage(shuffleId2)
+    var keepGoing = true
+    while (keepGoing) {
+      Thread.sleep(500)
+      keepGoing = shuffleStage1.latestInfo.attemptNumber() != 1
+    }
+    completeShuffleMapStageSuccessfully(0, 1, numPartitions)
+    keepGoing = true
+    while (keepGoing) {
+      Thread.sleep(500)
+      keepGoing = shuffleStage2.latestInfo.attemptNumber() != 1
+    }
+
+    completeShuffleMapStageSuccessfully(1, 1, numPartitions)
+    keepGoing = true
+    while (keepGoing) {
+      Thread.sleep(500)
+      keepGoing = resultStage.latestInfo.attemptNumber() != 1
+    }
+
+    assert(resultStage.latestInfo.numTasks == 2)
+  }
+
   test("SPARK-25341: retry all the succeeding stages when the map stage is indeterminate") {
     val (shuffleId1, shuffleId2) = constructIndeterminateStageFetchFailed()
 
     // Check status for all failedStages
     val failedStages = scheduler.failedStages.toSeq
     assert(failedStages.map(_.id) == Seq(1, 2))
     // Shuffle blocks of "hostC" is lost, so first task of the `shuffleMapRdd2` needs to retry.
+    // TODO: Asif THIS ASSERTION APPEARS TO BE WRONG. As the ShuffleMapStage is inDeterminate all
```

Comment:
Please remove the TODO or remove you name from the comment.

Reply:
done.